contracts/access/AccessControl.sol

```diff
@@ -133,7 +100,7 @@ abstract contract AccessControl is Context {
      * - the caller must have ``role``'s admin role.
      */
     function grantRole(bytes32 role, address account) public virtual {
-        require(hasRole(_roles[role].adminRole, _msgSender()), "AccessControl: sender must be an admin to grant");
+        require(AccessControl.hasRole(AccessControl.getRoleAdmin(role), _msgSender()), "AccessControl: sender must be an admin to grant");
```

Comment:
What is the reason to use the fully qualified `AccessControl` functions?

If we're making these functions virtual I don't see a reason not to respect their overrides here.

The same goes for the rest of this contract.

Reply:
I'd say consistency with other contract. In ERC721, we decided that view function should be virtual BUT that overloading them should not affect the parent behavior (ownerOf for example).

I don't see why we should use fully qualified functions in ERC721 but not in AccessControl

Reply:
Because ERC721 has more complicated internal invariants, and numbers have to add up properly while booleans like `hasRole` are trivial for us to support overriding. I know this doesn't feel like a consistent rule but I think it's an acceptable compromise.